toxcore/nat_traversal.h

```diff
@@ -0,0 +1,55 @@
+/* nat_traversal.h -- Functions to traverse a NAT (UPnP, NAT-PMP).
+ *
+ *  Copyright (C) 2016 Tox project All Rights Reserved.
+ *
+ *  This file is part of Tox.
+ *  Tox is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Tox is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Tox.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef NAT_TRAVERSAL_H
+#define NAT_TRAVERSAL_H
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <stdint.h>
+
+
+/**
+ * The protocol that will be used by the nat traversal.
+ */
+#if defined(HAVE_LIBMINIUPNPC) || defined(HAVE_LIBNATPMP)
```

Comment:
Conditional compilation is not really necessary here. Can you abstract the traversal stuff away and have just a single function instead of two? It's not nice that both TCP_server.c and network.c needs to know about the existence of these two libraries.